From 45a0e076d4575f2063d0b1bcc9beb7521cd264af Mon Sep 17 00:00:00 2001
From: Stefan Roese <sr@denx.de>
Date: Fri, 20 Dec 2019 13:18:38 +0100
Subject: [PATCH 6/9] WIP: net: DSA: mt7628-esw: Report link also on CPU port
 (eth0)

In the IOT mode (this is the only mode tested right now), lets also
report the link status on the CPU net device (e.g. eth0).

Signed-off-by: Stefan Roese <sr@denx.de>
---
 drivers/net/dsa/mt7628-esw.c | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/drivers/net/dsa/mt7628-esw.c b/drivers/net/dsa/mt7628-esw.c
index 9e25b1e671ce..c15942624f23 100644
--- a/drivers/net/dsa/mt7628-esw.c
+++ b/drivers/net/dsa/mt7628-esw.c
@@ -618,6 +618,8 @@ static irqreturn_t mt7628_esw_interrupt(int irq, void *_priv)
 
 		for (i = 0; i < MT7628_ESW_NUM_LEDS; i++) {
 			const struct dsa_port *dp = dsa_to_port(priv->ds, i);
+			const struct dsa_port *dp_cpu =
+				dsa_to_port(priv->ds, MT7628_ESW_CPU_PORT);
 
 			/* Skip disabled ports */
 			if (!priv->ports[i].enable)
@@ -630,6 +632,15 @@ static irqreturn_t mt7628_esw_interrupt(int irq, void *_priv)
 				netif_carrier_on(dp->slave);
 			else
 				netif_carrier_off(dp->slave);
+
+			/*
+			 * On the IOT mode (only PHY, no switch used) also
+			 * set link on the master net node (eth0)
+			 */
+			if (link & BIT(i))
+				netif_carrier_on(dp_cpu->slave);
+			else
+				netif_carrier_off(dp_cpu->slave);
 		}
 	}
 
-- 
2.29.2

